src/Processors/Transforms/Streaming/GlobalAggregatingTransformWithSubstream.cpp

```diff
@@ -30,17 +31,24 @@ SubstreamContextPtr GlobalAggregatingTransformWithSubstream::getOrCreateSubstrea
     auto substream_ctx = AggregatingTransformWithSubstream::getOrCreateSubstreamContext(id);
     if (params->emit_changelog && !substream_ctx->hasField())
     {
+        bool retract_enabled = false;
         substream_ctx->setField(
-            {std::make_shared<RetractedDataVariants>(),
+            {retract_enabled,
              /// Field serializer
-             [this](const std::any & field, WriteBuffer & wb, VersionType) {
-                 const auto & data = std::any_cast<const RetractedDataVariantsPtr &>(field);
-                 data->serialize(wb, params->aggregator);
+             [](const std::any & field, WriteBuffer & wb, VersionType version) {
```

Comment:
It may be a over engineering to attach serde functors to the field since we will need explicit cast it anyway in different places, we can't completely hide the underlying types. I would suggest avoiding this effort, cast on serde of Substream and explicity read / write binary for it appears more direct. We can do this in an other PR.

Reply:
I will do this in an other PR